aten/src/ATen/cuda/cub.cuh

```diff
@@ -45,17 +45,21 @@
 
 #ifdef USE_ROCM
 #define NO_ROCM(x)
+#define ROCM_HIPCUB(x) ::hipcub
 #else
 #define NO_ROCM(x) x
+#define ROCM_HIPCUB(x) x
 #endif
 
-#if !defined(USE_ROCM) && !CUB_SUPPORTS_NV_BFLOAT16()
+#if !CUB_SUPPORTS_NV_BFLOAT16() || \
+     (defined(USE_ROCM) && ROCM_VERSION >= 40500)
 
+#if !defined(USE_ROCM)
 namespace at_cuda_detail {
-// backport https://github.com/NVIDIA/cub/pull/306 for c10::BFloat16
+#endif
```

Comment:
Did you intend to remove the backport comment?  Not sure if it was still needed.

Reply:
Yes, it was intentional removed. The changes done in PR [#306](https://github.com/NVIDIA/cub/pull/306) cub is now supported in hipcub, so effectively backport is done.

Reply:
I think the comment was not referring to ROCm/hipcub, but instead to the below logic that's needed for CUDA/cub as well, so I think we should keep it?

Reply:
Comment was introduced by this commit: https://github.com/pytorch/pytorch/commit/b8dfb45ac282a48764c192ac7d27b7d80eed8b2b

Reply:
The comment was original introduced in the commit - [1bd22e2](https://github.com/pytorch/pytorch/commit/1bd22e28b318b15cbfe37ee1a112df9423aab328) PR#[58196](https://github.com/pytorch/pytorch/pull/58196), in the !USE_ROCM block of code since ROCm didnt support Bfloat16 type while doing radix sort. 

Also the cub PR#[306](https://github.com/NVIDIA/cub/pull/306) is mentioned few other places in the code which enables bfloat16 support for sort in cub lib, but backport was required only for this section where c10::BFloat16 dtype wasnt supported by ROCm for sort operator.

Now we have enabled this code block for ROCm.
[code](https://github.com/pytorch/pytorch/pull/71226/files#diff-b326bdf3f7652f753f72c742c5f65b4db7529ba317cea9c895723ddbc88ccab6R115-R119)

Reply:
I have corrected the code, since these changes are not required. Reverted and added back the comment too.